enzy_htp/structure/structure_operation/residue.py

```diff
@@ -111,3 +111,34 @@ def check_res_topology_error(
     # 3. check for any bond (polyline) from the target residue thread through any ring
     # 4. check for any ring from the target residue is threaded by any bond
     pass
+
+
+def closest_n_residues( 
+    target:Residue,
+    n:int,
+    method:str='centerofmass',
+    include_H:bool=False) -> List[Residue]:
+    """Find the closest n Residue() objects to a given Residue(). Complements a number of residue based functions.
+
+    Args:
+        target: The initial Residue() we are trying to find other Residue()'s near to.
+        n: The cutoff for top n Residue()'s that will be selected.
+        method: What metric should be used to rank Residue() distances? Default is 'centerofmass'.
+        include_H:Should Hydrogens be considered in various center of mass or other distance calculation? False by default.
+
+    Returns:
+        A List[Residue] containing the closest n residues.
+    """
+    target_coord = target.geom_center
+    distances = list()
+    for res in target.root().residues:
+        if res == target:
+            continue
+        
+        distances.append((mh.get_distance(target_coord, res.geom_center), res))
```

Comment:
Have mh being imported?

Reply:
This is a good point! It is not yet.

Reply:
Updated and fixed this. Validated with unittests